examples/src/main/scala/org/apache/spark/examples/sql/SparkSQLExample.scala

```diff
@@ -60,7 +60,8 @@ object SparkSQLExample {
 
   private def runBasicDataFrameExample(spark: SparkSession): Unit = {
     // $example on:create_df$
-    val df = spark.read.json("examples/src/main/resources/people.json")
+    val df = spark.read.json(
+      "file:///" + System.getProperty("user.dir") + "examples/src/main/resources/people.json")
```

Comment:
@srowen thanks for taking a look! Unfortunately, your suggestion doesn't work well in this case. I tried it on my local spark shell:

``` 
scala> val df = spark.read.json("file:examples/src/main/resources/people.json")
16/11/10 20:12:25 WARN datasources.DataSource: Error while looking for metadata directory.
java.lang.IllegalArgumentException: java.net.URISyntaxException: Relative path in absolute URI: file:examples/src/main/resources/people.json
  at org.apache.hadoop.fs.Path.initialize(Path.java:206)
  at org.apache.hadoop.fs.Path.<init>(Path.java:172)
  at org.apache.spark.sql.execution.datasources.DataSource$$anonfun$8.apply(DataSource.scala:283)
  at org.apache.spark.sql.execution.datasources.DataSource$$anonfun$8.apply(DataSource.scala:282)
  at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
  at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
  at scala.collection.immutable.List.foreach(List.scala:381)
  at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
  at scala.collection.immutable.List.flatMap(List.scala:344)
  at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:282)
  at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:152)
  at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:297)
  at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:250)
```

As you mentioned, the fix is not correct if we are not in spark homw directory. How about fixing the documentation to do the comments in spark directory?